src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -270,5 +272,63 @@ public static ObjectType UnderlyingGraphQLEntityType(IType type)
                         innerException: error);
             }
         }
+
+        /// <summary>
+        /// Generates the entity name from the GraphQL context.
+        /// </summary>
+        /// <param name="context">Middleware context.</param>
+        /// <returns></returns>
+        public static string GetDataSourceNameFromGraphQLContext(IMiddlewareContext context, RuntimeConfig runtimeConfig)
+        {
+            string rootNode = context.Selection.Field.Coordinate.TypeName.Value;
+            string dataSourceName;
+
+            if (string.Equals(rootNode, "mutation", StringComparison.OrdinalIgnoreCase) || string.Equals(rootNode, "query", StringComparison.OrdinalIgnoreCase))
+            {
+                // we are at the root query node - need to determine return type and store on context.
+                // Output type below would be the graphql object return type - Books,BooksConnectionObject.
+                IOutputType outputType = context.Selection.Field.Type;
+                string entityName = outputType.TypeName();
+
+                // Below is only needed if say we have an Array return type or items object for plural case.
+                ObjectType underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
+
+                // Example: CustomersConnectionObject - for get all scenarios.
+                if (QueryBuilder.IsPaginationType(underlyingFieldType))
+                {
+                    IObjectField subField = GraphQLUtils.UnderlyingGraphQLEntityType(context.Selection.Field.Type).Fields[QueryBuilder.PAGINATION_FIELD_NAME];
+                    outputType = subField.Type;
+                    underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
+                    entityName = underlyingFieldType.Name;
+                }
+
+                // if name on schema is different from name in config.
+                // Due to possibility of rename functionality, entityName on runtimeConfig could be different from exposed schema name.
+                if (GraphQLUtils.TryExtractGraphQLFieldModelName(underlyingFieldType.Directives, out string? modelName))
+                {
+                    entityName = modelName;
+                }
```

Comment:
for cosmos, we support an edge case where `@model` may not have a value inside it e.g. `@model(name="myEntity")`
Can you add a test/validate what happens when this is the case for cosmos? we don't want a null/empty value being set for `entityName`

Reply:
<img width="354" alt="image" src="https://github.com/Azure/data-api-builder/assets/124841904/4e8cc173-d448-498f-9a61-b9f5524b3cac">
this part of the code does verify it. I have in my multisource unit test though left the model name blank for the cosmos entity. and the above code ensures that we are not setting entityName to null or empty.

Reply:
Ah that's right. Thanks for the clarification!